<a href="https://colab.research.google.com/github/mykyto-maliy/data-analytics-in-it/blob/main/hw_1_1_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [ ]:
a = 101
b = 210
c = 111
d = 402

In [ ]:
result = ((a + b) * c) / d

print(result)

# я би округлив ще для бьютыфул виду ось так: print(round(result, 2))

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [ ]:
e = 11283
f = 89
g = e % f

print(g)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [ ]:
h =  e // f

print(h)

126


4. Запишіть у змінну `name` ваше імʼя.

In [10]:
name = 'Mykyto'

5. Запишіть у змінну `surname` ваше прізвище.

In [30]:
surname = 'Maliy'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [14]:
name_surname = name + ' ' + surname

print(name_surname)

# щоб було як у відповіді нижче (відповідь у лапках) треба просто написати змінну name_surname без print, я це розумію :)

Mykyto Maliy


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [18]:
print(name_surname.upper())

MYKYTO MALIY


In [20]:
name_surname.upper()

'MYKYTO MALIY'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [22]:
name_surname_upper = name_surname.upper()

name_surname_upper

'MYKYTO MALIY'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [24]:
name_surname[2]

'k'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [26]:
print(len(surname))

5


In [34]:
surname[0], surname[4]

('M', 'y')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [35]:
print(len(surname))

5


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [51]:
capital = 'Antananarivo'
l_capital = capital.lower()
count_capital = l_capital.count('a')
print(count_capital)

# я не зрозумів, чому нам дали це завдяння не даючи функцію count перед цим. Ок, я можу погуглити, але це дивно

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [56]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'

print(my_string.startswith('The'))
print(my_string.endswith('dog!'))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [76]:
new_string = my_string.split()
new_string_2 = '_'.join(new_string)
print(new_string_2)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`